arcaflow_plugin_sysbench/sysbench_plugin.py

```diff
@@ -58,10 +58,13 @@ def parse_output(output):
                 elif "Totaloperations" in key:
                     to, tops = value.split("(")
                     tops = tops.replace("persecond)", "")
-                    dictionary["Totaloperations"] = float(to)
+                    dictionary["Totaloperations"] = int(to)
                     dictionary["Totaloperationspersecond"] = float(tops)
                 elif "percentile" not in key and value.isnumeric():
-                    dictionary[key] = float(value)
+                    try:
+                        dictionary[key] = int(value)
+                    except ValueError:
+                        dictionary[key] = float(value)
```

Comment:
I was curious enough about this pattern to double-check `str.isnumeric()`.

It turns out that `"1.1".isnumeric()` is False, not True. It really is only looking for numerals... it seems this code was only treating *integers* as floating point values, while saving floating point values as strings. Your change just allows the integers to be stored as integers, making the `float(value)` branch unreachable. So a better way to do this (though it's not clear from the diffs why `*percentile*` keys were being handled as literals), might be something like:

```python
if "percentile" not in key:
    try:
        dictionary[key] = int(value)
    except ValueError:
        try:
            dictionary[key] = float(value)
        except ValueError:
            dictionary[key] = value
else:
    dictionary[key] = value
```

Reply:
It's a good observation. I wasn't happy with this either, but was trying to build on the existing code without too many changes.

> It turns out that "1.1".isnumeric() is False, not True.

This is interesting, and should imply that the original code shouldn't have worked. Maybe this path was never hit and it survived on the `else` condition simply casting the string to the value.

Your `try / except` method seems a lot more rational. I'll test it out.

Reply:
I reworked this, and it became clear from the test script that this is the better/correct approach. Thanks!

Reply:
Great. I wish I understood exactly what the intent of the "percentile" exclusion was. There's a similar `elif` below that does conversions for `"percentile"` values, although I'm not clear on the semantics of `dictionary == sysbench_output` vs the alternatives. Anyway, it seems like this ought to be the right thing to do in general! 😆

Reply:
The percentile exclusion was actually misplaced in this block, and I removed it with the update. This should only be in the block below where the latency values are aggregated. In the case of the percentile output, sysbench varies what we parse as the key based on the percentile selected in the input -- i.e. reporting `99th percentile : 1.234`. So for this case, we take the special action of parsing that key into a separate value so that we can report both the `percentile` and the `percentile_value` in the schema output.